### PR TITLE
[Promotion] Promotion position should not be null

### LINKED ISSUE #7694: [Promotion] Bug with nulling the priority
Steps to reproduce:
```gherkin
    @ui
    Scenario: Trying to remove priority from existing promotion
        Given there is a promotion "Christmas sale"
        And I want to modify this promotion
        When I remove its priority
        And I try to save my changes
        Then I should be notified that priority is required
```
Instead of notification the `HTTP 500 Error` is thrown. 

The question is how should it be fixed. Not null validation on a field? But during creation null is a valid value... Do you have any idea?

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | fixes #7694 |
| License         | MIT |

### CODE REVIEW

File: features/promotion/managing_promotions/promotion_validation.feature [hunk 1]
```diff
@@ -45,6 +45,14 @@ Feature: Promotion validation
         And this promotion should still be named "Christmas sale"
 
     @ui
+    Scenario: Trying to remove name from existing promotion
```
Comment (Zales0123) [inline features/promotion/managing_promotions/promotion_validation.feature]: `Trying to remove priority from existing promotion`

File: src/Sylius/Behat/Page/Admin/Promotion/UpdatePage.php [hunk 2]
```diff
@@ -25,6 +25,14 @@ class UpdatePage extends BaseUpdatePage implements UpdatePageInterface
     use ChecksCodeImmutability;
 
     /**
+     * @param int $priority
+     */
+    public function setItsPriority($priority = null)
```
Comment (Zales0123) [inline src/Sylius/Behat/Page/Admin/Promotion/UpdatePage.php]: It should be also placed in interface.

File: features/promotion/managing_promotions/promotion_validation.feature [hunk 3]
```diff
@@ -45,6 +45,14 @@ Feature: Promotion validation
...
+        Given there is a promotion "Christmas sale"
+        And I want to modify this promotion
```
Comment (GSadee) [inline features/promotion/managing_promotions/promotion_validation.feature]: `When`?

File: src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php [hunk 4]
```diff
@@ -118,6 +118,14 @@ public function iNameIt($name = null)
     }
 
     /**
+     * @When I remove its priority
+     */
+    public function iSetItsPriority()
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php]: `iRemoveItsPriority()`?

**Comment by lchrusciel** [comment]
Closing in favour of #7698

**Pull request closed without merging**